server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -924,6 +1015,135 @@ private void mergeMetadataRecords(MergeInfo info) throws AccumuloException {
     }
   }
 
+  private boolean needsFencingForDeletion(MergeInfo info, KeyExtent keyExtent) {
+    return needsFencingForSplit(info, keyExtent) || info.deleteOverlaps(keyExtent);
+  }
+
+  // Instead of splitting or chopping tablets for a delete we instead create ranges
+  // to exclude the portion of the tablet that should be deleted
+  private List<Range> createRangesForDeletion(TabletMetadata tabletMetadata,
+      final KeyExtent deleteRange) {
+    final KeyExtent keyExtent = tabletMetadata.getExtent();
```

Comment:
Not completely sure the following assumption is true, but think it might be.  Still looking through the code. If the assumption is true it would be good to add the validation.

```suggestion
   // its assumed that the delete range overlaps this range but does not contain it.
    Preconditions.checkArgument(!deleteRange.contains(tabletMetadata.getExtent()));

    final KeyExtent keyExtent = tabletMetadata.getExtent();
```